lib/msal-node/docs/migration.md

```diff
@@ -0,0 +1,92 @@
+
+**Q: Will you release a tool that helps me move my apps from ADAL to MSAL?**
+
+A: No. Differences between the libraries would require dedicating resources to development and maintenance of the tool that would otherwise be spent improving MSAL. However, we do provide the set of migration guides below to help you make the required changes in your application.
+
+**Q: How does MSAL work with AD FS?**
+
+A: MSAL for Node.js supports certain scenarios to authenticate against AD FS 2019. If your app needs to acquire tokens directly from earlier version of AD FS, you should remain on ADAL. Learn more.
+
+**Q: How do I get help migrating my application?**
+
+A: See the Migration guidance section of this article. If, after reading the guide for your app's platform, you have additional questions, you can post on Microsoft Q&A with the tag [azure-ad-adal-deprecation] or open an issue in library's GitHub repository. See the Languages and frameworks section of the MSAL overview article for links to each library's repo.
+
+## Differences between MSAL and ADAL for node.js
+For overall differences between ADAL.js and MSAL.you might want to read [Differences between MSAL.js and ADAL.js](https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-compare-msal-js-and-adal-js)
+
+### Difference in the Core API
+- The method to acquire tokens using client credentials is has been renamed in MSAL to `acquireTokenByClientCredentials` from `acquireTokenWithClientCredentials`
+
+- The methods to acquire tokens using the device code flow have been merged into `acquireTokenByDeviceCode` from the `acquireUserCode`, `acquireTokenWithDeviceCode` and `acquireToken` used to acquire a token via device code in MSAL.
+
+- The method to acquire tokens using a refresh token has been renamed to `acquireTokenByRefreshToken` in MSAL from `acquireTokenWithRefreshToken` in ADAL
+
+- The method to acquire tokens using a username and password pais has been renamed to `acquireTokenByUsernamePassword` in MSAL from `acquireTokenByUsernamePassword` in ADAL
+
+- The use of a certificate as a credential to acquire a token has been moved from the `acquireTokenWithClientCertificate` and has been included in as one of the modes of authentication in the `ConfidentialApplication` user agent client application.
+
+### Migrating from the AuthenticationContext to PublicClientApplication or ConfidentialClientApplication
+
+#### Constructing PublicClientApplication or ConfidentialClientAppliation
+When you use MSAL, you instantiate either a `PublicClientApplication` or a `ConfidentialClientApplication`. This object models your app identity and is used to make your requests through whichever flow you want. With this object you will configure your client idenitity, redirect URI, default authority, the log level and more.
```

Comment:
Can we please add code samples or link to existing `msal-node` docs where applicable?

Reply:
I have added links to existing documentation, please re-review and tell me if this matches your expectation